libs/fluid-elements/input/src/lib/input.ts

```diff
@@ -0,0 +1,183 @@
+ * @cssprop --fluid-input--foreground-key-focus - Controls the foreground focus color for the default state.
+ * @cssprop --fluid-input--background-key-focus - Controls the background focus color for the default state.
+ * @cssprop --fluid-input--border-key-focus - Controls the border focus color for the default state.
+ * @cssprop --fluid-input--foreground-negative - Controls the foreground color for the negative state.
+ * @cssprop --fluid-input--border-negative - Controls the border color for the negative state.
+ * @cssprop --fluid-input--foreground-negative-hover - Controls the foreground hover color for the negative state.
+ * @cssprop --fluid-input--border-negative-hover - Controls the border hover color for the negative state.
+ * @cssprop --fluid-input--foreground-negative-focus - Controls the foreground focus color for the negative state.
+ * @cssprop --fluid-input--border-negative-focus - Controls the border focus color for the negative state.
+ * @cssprop --fluid-input--foreground-disabled - Controls the foreground color for the disabled state.
+ * @cssprop --fluid-input--background-disabled - Controls the background color for the disabled state.
+ * @cssprop --fluid-input--border-disabled - Controls the border color for the disabled state.
+ * @cssprop --fluid-input--placeholder - Controls the placeholder color.
+ * @cssprop --fluid-input--hint - Controls the hint color.
+ */
+@customElement('fluid-input')
+export class FluidInput extends LitElement {
+  /** Styles for the input component */
+  static get styles(): CSSResult {
+    return inputStyles;
+  }
+
+  /**
+   * Defines the aria label of the input input field.
+   * @attr
+   * @type string
+   */
+  @property({ type: String, reflect: true })
+  labelposition: FluidInputLabelPosition = `top`;
+
+  /**
+   * Defines the hint of the input input field.
+   * @attr
+   * @type string
+   */
+  @property({ type: String, reflect: true })
+  hint: string = ``;
+
+  /**
+   * @internal Defines if the input should be blurred
+   * Used in the fluid-combo-box to keep the input
+   * focused when clicking in the popover
+   */
+  _preventBlur = false;
+
+  /**
+   * Defines whether the input is disabled.
+   * Used for styling the input container
+   */
+  @property({ type: Boolean, attribute: false })
+  private _disabled = false;
+
+  /**
+   * Defines whether the input is focused.
+   * Used for styling the container
+   */
+  @property({ type: Boolean, attribute: false })
+  private _focus = false;
+
+  /**
+   * Validate slotted element(s) and add event listeners
+   * for focusing and blurring an input
+   */
+  private _handleSlotChange({ target }: { target: HTMLSlotElement }): void {
+    const slottedElements = target.assignedElements({
+      flatten: true,
+    });
+
+    // Only accept one element per slot
+    if (slottedElements.length > 1) {
+      throw new Error(
+        `The fluid-input only takes one input element in the default slot.`,
+      );
```

Comment:
@ffriedl89 do we really want to throw an error or is logging enough?

Reply:
I think with a changed API where the form field is extracted we can add a warning/error log. Throwing an error is usually not the go to i'd say. For now I wouldn't bother.